leapp/libraries/stdlib/__init__.py

```diff
@@ -32,3 +38,51 @@ def call(args, split=True):
     if split:
         return r.splitlines()
     return r
+
+
+def _logging_handler((fd, fd_type), buffer):
+    if os.getenv('LEAPP_DEBUG', '0') == '1':
+        if fd_type == call.STDOUT:
+            sys.stdout.write(buffer)
+        else:
+            sys.stderr.write(buffer)
+
+
+def new_call(args):
+    """
+    Call the Leapp's stdlib _call
+
+    :param args: Command to execute
+    :return: stdout output, 'utf-8' decoded
+    """
+    _id = str(uuid.uuid4())
+    result_data = None
+    try:
+        create_audit_entry('process-start', {'id': _id, 'parameters': args})
+        result_data = call._call(args, callback_raw=_logging_handler)
+    finally:
+        create_audit_entry('process-end', _id)
+        create_audit_entry('process-result', {'id': _id, 'parameters': args, 'result': result_data})
+    return result_data
+
+
+def new_checked_call(args, split=False):
+    """
+    Call the audited call and check result, split the result into multiple lines if requested
+
+    :param args: Command to execute
+    :param split: Split the output on newlines
+    :return: stdout output, 'utf-8' decoded, split by lines if split=True
+    """
+    result = new_call(args)
+    if result['exit_code'] != 0:
+        failed_command = "Command: " + str(args[0])
+        command_stdout = "Result stdout: " + result['stdout']
+        error_message = " stderr: " + result['stderr']
+        raise CommandError(
```

Comment:
No, no, this is no good, do not use CommandError here. This is not for this purpose. It was meant to be used Leapp tools CLI handling.

The idea of defining a CalledProcessError was much better, since that's what people are used to.
```python
class CalledProcessError(LeappError):
    def __init__(self, message, command, result):
        super(CalledProcessError, self).__init__(message)
        self._result = result
        self.command = command
    
    @property
    def stdout(self):
        self._result.get('stdout')

    @property
    def stderr(self):
        self._result.get('stderr')

    @property
    def exit_code(self):
        self._result.get('exit_code')

    @property
    def signal(self):
        self._result.get('signal')

    @property
    def pid(self):
        self._result.get('pid')
```
Something along these lines. This way you can get the data and don't try to consider stderr to be an error message. This is not a correct assumption. This can be tons of text....